src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -45,25 +45,24 @@ def algorithm(self, parameters):
             logging.error("Failed to fetch agent configuration from the auxiliary DB")
             return
 
-        if isDrainMode(self.config):
-            # check to see if the agent hit any speed drain thresholds
-            thresholdsHit = self.checkSpeedDrainThresholds()
-            if thresholdsHit:
-                logging.info("Updating agent configuration for speed drain...")
-                self.updateAgentSpeedDrainConfig(thresholdsHit)
-            # now collect drain statistics
-            try:
+        try:
+            if isDrainMode(self.config):
```

Comment:
@goughes Erik, as we quickly discussed in the meeting, `isDrainMode` is either using the cache or making a call to reqmgr_aux db again (as done a couple of lines above - getWMAgentConfig).
I think we can replace this line by what we have here:
https://github.com/dmwm/WMCore/blob/dfca72872e8994c761e33c5289c7bbf74e6604fe/src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py#L196

any thoughts?

Reply:
That should work. The only thing different in ReqMgrAux is they check for the keys first:
https://github.com/dmwm/WMCore/blob/36970485a95fe305fd767458c2e32b897a0885a3/src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py#L194

But, on the off chance that those keys don't exist, the KeyError will be caught and reported.